storage/storage.go

```diff
@@ -348,3 +395,205 @@ func EncodeName(name string) string {
+		if strings.HasPrefix(file.Name, ".pdb") || strings.HasPrefix(file.Name, ".debug") || strings.Contains(file.Name, modVersion) {
+			continue
+		}
+
+		if strings.Contains(file.Name, "LinuxServer") {
+			err = WriteZipFile(file, "LinuxServer", zipWriterLinuxServer)
+
+			if err != nil {
+				fmt.Println(err)
+				return false, ""
+			}
+
+			LinuxServer = true
+		}
+
+		if strings.Contains(file.Name, "WindowsServer") {
+			err = WriteZipFile(file, "WindowsServer", zipWriterWin64Server)
+
+			if err != nil {
+				fmt.Println(err)
+				return false, ""
+			}
+
+			Win64Server = true
+		}
+
+		if strings.Contains(file.Name, "WindowsNoEditor") {
+			err = WriteZipFile(file, "WindowsNoEditor", zipWriterWin64Client)
+
+			if err != nil {
+				fmt.Println(err)
+				return false, ""
+			}
+
+			Win64Client = true
+		}
+	}
+
+	// Make sure to check the error on Close.
+	err = zipWriterLinuxServer.Close()
+	if err != nil {
+		fmt.Println(err)
+		return false, ""
+	}
+	err = zipWriterWin64Server.Close()
+	if err != nil {
+		fmt.Println(err)
+		return false, ""
+	}
+	err = zipWriterWin64Client.Close()
+	if err != nil {
+		fmt.Println(err)
+		return false, ""
+	}
+
+	//Write to mod_link and upload new smaller smod file
+	if LinuxServer {
+		key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-LinuxServer-"+modVersion)
+		platform := "LinuxServer"
+
+		err = WriteModLink(ctx, key, versionID, platform, bufLinuxServer)
+		if err != nil {
+			fmt.Println(err)
+			return false, ""
+		}
+	}
+
+	if Win64Server {
+		key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-Win64Server-"+modVersion)
+		platform := "Win64Server"
+
+		err = WriteModLink(ctx, key, versionID, platform, bufWin64Server)
+
+		if err != nil {
+			fmt.Println(err)
+			return false, ""
+		}
+	}
+	if Win64Client {
+		key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-WindowsNoEditor-"+modVersion)
+		platform := "WindowsNoEditor"
+
+		err = WriteModLink(ctx, key, versionID, platform, bufWin64Client)
+
+		if err != nil {
+			fmt.Println(err)
+			return false, ""
+		}
+	}
+
+	key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-Combined-"+modVersion)
+	platform := "Combined"
+
+	combined := bytes.NewBuffer(body)
+
+	err = WriteModLink(ctx, key, versionID, platform, combined)
+
+	if err != nil {
+		fmt.Println(err)
+		return false, ""
+	}
+
+	key = fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-WindowsNoEditor-"+modVersion)
+
+	return true, key
+}
+
+func WriteZipFile(file *zip.File, platform string, zipWriter *zip.Writer) error {
+	fileName := strings.ReplaceAll(file.Name, platform+"/", "")
+	zipFile, err := zipWriter.Create(fileName)
+
+	if err != nil {
+		fmt.Println(err)
+		return fmt.Errorf("failed to create zip: %w", err)
+	}
+
+	rawFile, err := file.Open()
+
+	if err != nil {
+		fmt.Println(err)
+		return fmt.Errorf("failed to open zipped file: %w", err)
+	}
+
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(rawFile)
+
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Please remove the debug println as you return an error with context

Reply:
https://github.com/satisfactorymodding/smr-api/pull/12/commits/74cbf9e7fb628b477e7bb2459a0f641867ec9415